js/a11y/browserFocus.js

```diff
@@ -78,8 +79,19 @@ export default class BrowserFocus extends Backbone.Controller {
     if (!$focusable.length) {
       return;
     }
+    const $closestFocusable = $element.closest(config._options._tabbableElements);
     // Force focus for screen reader enter / space press
-    $focusable[0].focus();
+    if ($closestFocusable[0] !== document.activeElement) {
+      // Focus on the nearest focusable element if not already with focus
+      $closestFocusable[0].focus();
+    }
+    if (!config._options._isClickDelayedAfterFocusEnabled) return;
+    // Add a small delay to each click to allow screen readers to process focus
+    event.preventDefault();
+    event.stopImmediatePropagation();
+    _.delay(() => {
```

Comment:
would we be better making use of native `setTimeout` here if we are moving away from underscore?

Reply:
force of habit, sorry